### PR TITLE
Add Contributor Guidelines, User Guide, and Pull Request Template

### PR DESCRIPTION
**Pull request by IanHoang**
### Description
For the last few months, there's been some confusion about contributing to the workloads repository. This PR streamlines the process and adds details related to how to contribute to the workloads repository, establishes new PR best practices, and how to backport changes appropriately. 

### Issues Resolved
https://github.com/opensearch-project/opensearch-benchmark-workloads/issues/419
https://github.com/opensearch-project/opensearch-benchmark/pull/583#discussion_r1701920659

### Backport to Branches:
- [x] 6
- [x] 7
- [x] 1
- [x] 2
- [x] 3

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+- If the change is applicable to several branches, create the feature branch based off of `main` branch and make the changes there.
+
+## Test changes
+
+Once you've made your change in your feature branch, we recommend testing it locally and with integration tests via your forked OpenSearch Benchmark repository.
+
+### Testing changes locally
+It's recommended to test your change locally by performing the following:
+1. Set up or use an existing OpenSearch cluster to test your changes against.
+2. Based on the major version of the test cluster, cherry-pick the commit(s) with your change to the corresponding major version branch in your forked workloads repository. For example, if you're testing against an OpenSearch 2.X.X cluster, cherry-pick the changes from the feature branch to `2` branch.
```
Comment (OVI3D0) [inline CONTRIBUTING.md:32]: did you mean the forked opensearch benchmark repo?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+- Ensure you are using the workloads repository that you committed your changes in. To enforce this, provide the path to your repository via the `--workloads-repository` parameter.
+- Alternatively, you can force OSB to use a specific branch by specifying the distribution version of your OpenSearch cluster via the `--distribution-version` parameter. To build on the example from the previous step, to ensure you are using branch `2`, set `--distribution-version=2.0.0` in the OpenSearch Benchmark command.
+
+### Testing changes with integration tests
+
+To ensure that there are no other breaking changes, we recommend testing with your forked OpenSearch Benchmark repository Github Actions.
+
+**Prerequisites:**
+
+Before we can run integration tests, we recommend dedicating a remote branch that you can repeatedly use to run integration tests on your forked workloads repository. You only need to perform these actions once.
```
Comment (OVI3D0) [inline CONTRIBUTING.md:45]: Did you mean the forked opensearch benchmark repository?

File: USER_GUIDE.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
+# OpenSearch Benchmark Workloads User Guide
+
+OpenSearch Benchmark (OSB) comes packaged with official workloads that can be found throughout this repository. This guide provides a brief overview of the organizsation of this repository as well as how OSB uses these workloads.
+
+### Contents
+- [What do branches represent](#what-do-the-numbered-branches-represent)
+- [How does OSB select which branch to use](#how-does-osb-select-which-branch-to-use)
+- [Force OSB to use a Specific Branch](#force-osb-to-use-a-specific-branch)
```
Comment (OVI3D0) [inline USER_GUIDE.md:8]: Did you plan to add more to this guide? Not sure if these links are all that helpful otherwise haha

**Review by OVI3D0 — COMMENTED** [review]
IMO this guide is very useful, and would really help someone who is new to the workloads repository (like myself) to get up and running without much confusion. Thanks for this!

File: CONTRIBUTING.md
Comment (OVI3D0) [inline CONTRIBUTING.md:1]: Maybe add a note or link mentioning signing commits and DCO?

File: CONTRIBUTING.md (hunk 2 above)
Comment (IanHoang) [reply CONTRIBUTING.md:45]: Ah, you're right. I'll rephrase it to:
```
we recommend dedicating a remote branch in your forked OpenSearch Benchmark repository. You will use this branch whenever you need to run integration tests against changes in your forked workloads repository. 
```

File: USER_GUIDE.md (hunk 3 above)
Comment (IanHoang) [reply USER_GUIDE.md:8]: That's a good point, I just added it in case we ever expand upon this guide. I'm fine with keeping it or discarding it.

File: CONTRIBUTING.md
Comment (IanHoang) [reply CONTRIBUTING.md:1]: Good call out!

**Review by gkamat — CHANGES_REQUESTED** [review]

File: USER_GUIDE.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
+# OpenSearch Benchmark Workloads User Guide
+
+OpenSearch Benchmark (OSB) comes packaged with official workloads that can be found throughout this repository. This guide provides a brief overview of the organizsation of this repository as well as how OSB uses these workloads.
```
Comment (gkamat) [inline USER_GUIDE.md:3]: comes packaged with several workloads that are included in this repository.

File: USER_GUIDE.md [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+### Contents
+- [What do branches represent](#what-do-the-numbered-branches-represent)
+- [How does OSB select which branch to use](#how-does-osb-select-which-branch-to-use)
+- [Force OSB to use a Specific Branch](#force-osb-to-use-a-specific-branch)
+
+### What do the numbered branches represent?
+
+Don't worry, these numbers are not the same [numbers](https://lostpedia.fandom.com/wiki/The_Numbers) in the series [Lost](https://en.wikipedia.org/wiki/Lost_(2004_TV_series)). Each branch -- `main`, `6`, `7`, `1`, `2`, `3` -- is associated with a specific major version of OpenSearch or Elasticsearch and contains variations of each workload.
```
Comment (gkamat) [inline USER_GUIDE.md:12]: contain variations of relevant workloads appropriate for the respective branches.  Note that versions `6` and `7` can be thought of as "older" versions of OpenSearch, based on an historical version-numbering oddity.

**Review by gkamat — CHANGES_REQUESTED** [review]

**Review by gkamat — APPROVED** [review]
